### PR TITLE
Add support for generic type for returned generated key

### PR DESCRIPTION
**Pull request by stefanbacon**
Add support for generic type for returned generated key

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import MimaSettings.mimaSettings
 
-lazy val _version = "3.2.2"
+lazy val _version = "3.2.3"
```
Comment (seratch) [inline build.sbt]: You don't need to modify this part. We, the library maintainers do this.

File: scalikejdbc-core/src/main/scala/scalikejdbc/DBSession.scala [hunk 2]
```diff
@@ -617,40 +617,40 @@ trait DBSession extends LogSupport with LoanPattern with AutoCloseable {
...
+    var generatedKey: T = null.asInstanceOf[T]
     val before = (stmt: PreparedStatement) => {}
     val after = (stmt: PreparedStatement) => {
       val rs = stmt.getGeneratedKeys
       while (rs.next()) {
         generatedKeyFound = true
         generatedKey = key match {
           case name: String => try {
-            rs.getLong(name)
+            rs.getObject(name).asInstanceOf[T]
```
Comment (seratch) [inline scalikejdbc-core/src/main/scala/scalikejdbc/DBSession.scala]: I am not sure if using getObject and casting the value is the best way yet. I would like to know what other libraries do for this purpose. Do you have any pieces of evidence for it?

Comment (faresbessrour) [reply scalikejdbc-core/src/main/scala/scalikejdbc/DBSession.scala]: @stefanbacon 

```scala
rs.get[T](name)
```

Maybe this works?

**Comment by seratch** [comment]
As far as I know, supporting any types for returned generated keys is not so simple. Let me close this pull request due to inactivity but if someone will come up with a solution with sufficient test code, I'm happy to review it.

**Pull request closed without merging**